pkg/templates/clusterconfig_funcs.go

```diff
@@ -36,6 +35,12 @@ func (t *TemplateResolver) fromClusterClaim(claimname string) (string, error) {
 
 	getObj, getErr := dclient.Get(context.TODO(), claimname, metav1.GetOptions{})
 	if getErr != nil {
+		if k8serrors.IsNotFound(getErr) {
+			// Add to the referenced objects if the object isn't found since the consumer may want to watch the object
+			// and resolve the templates again once it is present.
+			t.addToReferencedObjects(clusterClaimAPIVersion, "ClusterClaim", "", claimname)
+		}
+
```

Comment:
Why not add it to the referenced objects a single time up front before retrieving the dynamic client?

Reply:
I didn't want to add the referenced object in case it was invalid in some other way than it not being found. This is to protect in case the user input in the template would cause the kubernetes-dynamic-watches library to error or something (e.g. an empty string for the name).